packages/ODEs/src/app.ts

```diff
@@ -1,654 +1,870 @@
+  {label: `${CONTROL_EXPR.TAGS}: `, type: 'keyword', info: INFO.TAGS},
+  {label: `${CONTROL_EXPR.DESCR}: `, type: 'keyword', info: INFO.DESCR},
+  {label: `${CONTROL_EXPR.DIF_EQ}:\n  `, type: 'keyword', info: INFO.DIF_EQ},
+  {label: `${CONTROL_EXPR.EXPR}:\n  `, type: 'keyword', info: INFO.EXPR},
+  {label: `${CONTROL_EXPR.ARG}: `, type: 'keyword', info: INFO.ARG},
+  {label: `${CONTROL_EXPR.INITS}:\n  `, type: 'keyword', info: INFO.INITS},
+  {label: `${CONTROL_EXPR.PARAMS}:\n  `, type: 'keyword', info: INFO.PARAMS},
+  {label: `${CONTROL_EXPR.CONSTS}:\n  `, type: 'keyword', info: INFO.CONSTS},
+  {label: `${CONTROL_EXPR.TOL}: `, type: 'keyword', info: INFO.TOL},
+  {label: `${CONTROL_EXPR.LOOP}:\n  `, type: 'keyword', info: INFO.LOOP},
+  {label: `${CONTROL_EXPR.UPDATE}:  `, type: 'keyword', info: INFO.UPDATE},
+  {label: `${CONTROL_EXPR.OUTPUT}:\n  `, type: 'keyword', info: INFO.OUTPUT},
+  {label: `${CONTROL_EXPR.COMMENT}: `, type: 'keyword', info: INFO.COMMENT},
 ];
 
 /** Control expressions completion utilite */
 function contrCompletions(context: any) {
-  let before = context.matchBefore(/[#]\w*/)
-  if (!context.explicit && !before) return null
+  const before = context.matchBefore(/[#]\w*/);
+
+  if (!context.explicit && !before) return null;
   return {
     from: before ? before.from : context.pos,
     options: completions,
-    validFor: /^\w*$/
-  }
+    validFor: /^\w*$/,
+  };
 }
 
 /** Return options of line chart */
-function lineChartOptions(colNames: string[]): Object {
+function getLineChartOptions(colNames: string[]): Object {
   const count = colNames.length;
-
   return {
     xColumnName: colNames[0],
-    yColumnNames: (count > 1) ? colNames.slice(1) : colNames[0],     
+    yColumnNames: (count > 1) ? colNames.slice(1).filter((name) => name !== STAGE_COL_NAME) : colNames[0],
     showTitle: true,
-    sharex: true, 
+    sharex: true,
     multiAxis: count > MAX_LINE_CHART,
-    multiAxisLegendPosition: "RightTop",
+    multiAxisLegendPosition: 'RightTop',
+    segmentColumnName: colNames.includes(STAGE_COL_NAME) ? STAGE_COL_NAME: null,
   };
 }
 
-/** Run solver application */
-export async function runSolverApp() {
+/**  String to value */
+const strToVal = (s: string) => {
+  const num = Number(s);
 
-  /** Get JS-script for solving the current IVP */
-  const exportToJS = async () => {
-    try {
-      const ivp = getIVP(editorView.state.doc.toString());
-      const scriptText = getScriptLines(ivp).join('\n');      
-      const script = DG.Script.create(scriptText);
+  if (!isNaN(num))
+    return num;
 
-      // try to call computations - correctness check
-      const params = getScriptParams(ivp);    
-      const call = script.prepare(params);
-      await call.call();
+  if (s === 'true')
+    return true;
 
-      const sView = DG.ScriptView.create(script);
-      grok.shell.addView(sView);
+  if (s === 'false')
+    return false;
+
+  return s;
```

Comment:
I think you can make this strToVal function in one string - something like
`return !isNan(+s) ? num : s === 'true' ? true : s === 'false' ? false : s`

Reply:
and just make it 2 lines long for the clarity

Reply:
Done